Code/GraphMol/MolAlign/AlignMolecules.cpp

```diff
@@ -112,6 +112,85 @@ double getBestRMS(ROMol &probeMol, ROMol &refMol, int probeId, int refId,
   return bestRMS;
 }
 
+double CalcRMS(ROMol &prbMol, const ROMol &refMol, int prbCid, int refCid,
+               const std::vector<MatchVectType> &map, int maxMatches,
+               const RDNumeric::DoubleVector *weights) {    
+  std::vector<MatchVectType> matches = map;
+  if (matches.empty()) {
+    bool uniquify = false;
+    bool recursionPossible = true;
+    bool useChirality = false;
+    bool useQueryQueryMatches = false;
+
+    SubstructMatch(refMol, prbMol, matches, uniquify, recursionPossible,
+                   useChirality, useQueryQueryMatches, maxMatches);
+
+    if (matches.empty()) {
+      throw MolAlignException(
+          "No sub-structure match found between the reference and probe mol");
+    }
+
+    if (matches.size() > 1e6) {
+      std::string name;
+      prbMol.getPropIfPresent(common_properties::_Name, name);
+      std::cerr << "Warning in " << __FUNCTION__ << ": " << matches.size()
+                << " matches detected for molecule " << name << ", this may "
+                << "lead to a performance slowdown.\n";
+    }
+  }
+  
+  unsigned int msize = matches[0].size();
+  const RDNumeric::DoubleVector *wts;
+  if (weights) {
+    PRECONDITION(msize == weights->size(), "Mismatch in number of points");
+    wts = weights;
+  } else {
+    wts = new RDNumeric::DoubleVector(msize, 1.0);
+  }
+
+  double bestRMS = 1.e300;
+  MatchVectType &bestMatch = matches[0];
+  for (auto &matche : matches) {
+    RDGeom::Point3DConstPtrVect refPoints, prbPoints;
+    const Conformer &prbCnf = prbMol.getConformer(prbCid);
+    const Conformer &refCnf = refMol.getConformer(refCid);
+    
+    MatchVectType::const_iterator mi;
+    for (mi = matche.begin(); mi != matche.end(); mi++) {
+      prbPoints.push_back(&prbCnf.getAtomPos(mi->first));
+      refPoints.push_back(&refCnf.getAtomPos(mi->second));
+    }
+    
+    unsigned int npt = refPoints.size();
+    PRECONDITION(npt == prbPoints.size(), "Mismatch in number of points");
+    double ssr = 0.; 
+    
+    const RDGeom::Point3D *rpt, *ppt;
+    for (unsigned int i = 0; i < npt; i++) {
+      rpt = refPoints[i];
+      ppt = prbPoints[i];
+      double wt =  (*wts)[i];
+ 
+      ssr += ((ppt->x) - (rpt->x)) * ((ppt->x) - (rpt->x)) * wt;
```

Comment:
The next three lines can be replaced with:
`ssr += wt*(ppt-rpt).lengthSq()`

Reply:
Aah yes, thank you! 
I wanted to use it first, but somehow forgot to dereference a pointer to the point, got some errors, started thinking that I am wrongly understanding usage of math operators with points and went back to the explicit distance computation.